appserver/tests/payara-samples/pom.xml

```diff
@@ -53,24 +52,35 @@
         <module>classpath</module>
     </modules>
 
-    <dependencies>
-        <!-- DO NOT ADD DEPENDENCIES -->
-    </dependencies>
+    <repositories>
+        <!-- omnifaces jacc-provider is not in any other repository -->
+        <repository>
+            <id>payara-eclipse</id>
+            <name>Payara Eclipse</name>
+            <url>https://raw.github.com/payara/Payara_PatchedProjects/eclipse</url>
+            <releases>
+                <enabled>true</enabled>
+            </releases>
+            <snapshots>
+                <enabled>false</enabled>
+            </snapshots>
+        </repository>
+    </repositories>
 
     <dependencyManagement>
         <dependencies>
             <dependency>
                 <groupId>fish.payara.api</groupId>
                 <artifactId>payara-bom</artifactId>
-                <!-- A workaround for case when module redefines its version (rest-monitoring-war) -->
-                <version>${project.parent.version}</version>
+                <version>${project.version}</version>
```

Comment:
Yes, using `${payara.version}` is the best approach here.

Then almost all of the dependency management section is then unnecessary, with exception to internal module references.

Reply:
Partially true, except the fact that bom brings compile scope to anything. Real time scope then depends on the "root" artifact. But even with these rules it helped much :)